### PR TITLE
Point-Reactor Kinetics Module

### PR DESCRIPTION
**Pull request by Naktakala**
# Introducing the Point-Reactor Kinetics module

## Highlights
- Added a new module `prk`, Point-Reactor Kinetics, with a `TransientSolver`.
  The new solver showcases the static registration paradigm that I want to shift
  towards.
- Changes to some macros in `chi_log_exceptions.h`, adding more functionality
  to `ChiLogicalError` and `ChiInvalidArgument`.
- Added a member to `InputParameters` called `system_ignored_param_names_`, 
  allowing us to filter out some system-level parameters.
- Added more comments to `InputParameters`.
- Made some changes to `ChiObjectMaker`, this is still a WIP.

### CODE REVIEW

File: modules/PointReactorKinetics/doc/rpk_TransientSolver_whitepaper.tex [hunk 1]
```diff
@@ -0,0 +1,371 @@
+\documentclass[11pt,letterpaper,notitlepage]{article}
```
Comment (ragusa) [inline modules/PointReactorKinetics/doc/rpk_TransientSolver_whitepaper.tex:1]: can you email me the PDF file? Thanks

File: modules/PointReactorKinetics/doc/rpk_doc.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+\defgroup prk Point Reactor Kinetics
+\ingroup LuaModules
+
+This module concerns itself with the solution of the Point-Reactor Kinetics 
+equations:
+\f[
+\label{Eq:1}
+\frac{dn}{dt} = \frac{\beta_{eff} (\rho(t)-1)}{\Gamma_0} n(t) +
```
Comment (ragusa) [inline modules/PointReactorKinetics/doc/rpk_doc.md]: Why is MGT \Gamma. I am more used to \Lambda

**Review by ragusa — APPROVED** [review]

File: modules/PointReactorKinetics/doc/rpk_TransientSolver_whitepaper.tex (hunk 1 above)
Comment (Naktakala) [reply modules/PointReactorKinetics/doc/rpk_TransientSolver_whitepaper.tex:1]: Will do.

File: modules/PointReactorKinetics/doc/rpk_doc.md (hunk 2 above)
Comment (Naktakala) [reply modules/PointReactorKinetics/doc/rpk_doc.md]: Wow! How did that sneak in there... Good catch thank you. It's a lambda in the pdf.

**Pull request merged**